lib/Doctrine/ORM/Query/SqlWalker.php

```diff
@@ -1125,15 +1125,18 @@ public function walkJoin($join)
                 $class      = $this->em->getClassMetadata($joinDeclaration->abstractSchemaName);
                 $dqlAlias   = $joinDeclaration->aliasIdentificationVariable;
                 $tableAlias = $this->getSQLTableAlias($class->table['name'], $dqlAlias);
-                $condition = '(' . $this->walkConditionalExpression($join->conditionalExpression) . ')';
+                $conditions = [];
+
+                if ($join->conditionalExpression) {
```

Comment:
Looks like a valid fix, but unrelated to this PR.
From what I could quickly see, you're handling the dis-jointed class support, such as: `SELECT u FROM User u JOIN Address a ON a.user_id = u.id`